src/ajax.ts

```diff
@@ -6,27 +6,42 @@
 /**
  * Created by Samuel Gratzl on 04.08.2014.
  */
-import {list as listPlugins} from './plugin';
 import {offline as isOffline, server_url, server_json_suffix} from '.';
 
-/**
- * interface for the ajax adapter
- */
-export interface IAjaxAdapter {
-  send(url: string, data: any, method : string, expectedDataType : string): Promise<any>;
+class AjaxError extends Error {
+  constructor(public response: any) {
+    super(response.statusText);
+  }
 }
 
-var _impl : Promise<IAjaxAdapter> = null;
-
-function getConnector() {
-  if (_impl != null) {
-    return _impl;
+function checkStatus(response) {
+  if (response.ok) {
+    return response;
+  } else {
+    throw new AjaxError(response);
   }
-  const adapters = listPlugins('ajax-adapter');
-  var adapter = adapters[0];
-  return _impl = adapter.load().then((p) => <IAjaxAdapter>p.factory());
 }
 
+function parseType(expectedDataType: string, response) {
+  expectedDataType = expectedDataType.trim().toLowerCase();
+  switch (expectedDataType) {
+    case 'json':
+    case 'application/json':
+      return response.json();
+    case 'text':
+    case 'text/plain':
+      return response.text();
+    case 'blob':
+      return response.blob();
+    case 'formdata':
+      return response.formData();
+    case 'arraybuffer':
+      return response.arrayBuffer();
+    default:
+      console.warn('unknown expected data type: ' + expectedDataType + ' returning blob');
+      return response.blob();
+  }
+}
```

Comment:
If this is for our own use, do we need to support so many formats? And if someone specifies a type we don't recognize, would be be better to fail explicitly now, rather than returning a result they might or might not be able to handle... when the error finally does break something it will be farther from the point where the mistake was made?